libs/langchain-community/src/llms/arcjet.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  LLM,
+  BaseLLM,
+  type BaseLLMCallOptions,
+} from "@langchain/core/language_models/llms";
+import type {
+  ArcjetSensitiveInfoType, RedactOptions
+} from "@arcjet/redact";
+import {
+  redact,
+} from "@arcjet/redact";
+
+type DetectSensitiveInfoEntities<T> = (tokens: string[]) => Array<ArcjetSensitiveInfoType | T | undefined>;
+type ValidEntities<Detect> = Array<undefined extends Detect ? ArcjetSensitiveInfoType : Detect extends DetectSensitiveInfoEntities<infer CustomEntities> ? ArcjetSensitiveInfoType | CustomEntities : never>;
+
+export interface ArcjetRedactOptions<Detect> extends BaseLLMCallOptions {
+  llm: BaseLLM;
+  entities?: ValidEntities<Detect>;
+  contextWindowSize?: number;
+  detect?: Detect;
+  replace?: (entity: ValidEntities<Detect>[number]) => string | undefined;
+}
+
+export class ArcjetRedact<const Detect extends DetectSensitiveInfoEntities<CustomEntities> | undefined, const CustomEntities extends string> extends LLM {
+  static lc_name() {
+    return "ArcjetRedact";
+  }
+
+  llm: BaseLLM;
+
+  entities?: ValidEntities<Detect>;
+
+  contextWindowSize?: number;
+
+  detect?: Detect;
+
+  replace?: (entity: ValidEntities<Detect>[number]) => string | undefined;
+
+  constructor(options: ArcjetRedactOptions<Detect>) {
+    super(options);
+
+    if (!options.llm) {
```

Comment:
Can we rely on TypeScript for some of these?

Reply:
Yeah, I've removed all of the ones that aren't handled by TS